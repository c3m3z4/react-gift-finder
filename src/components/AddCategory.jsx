import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                name=" "
                value={inputValue}
                onChange={onInputChange}
                placeholder="Buscar Gifs"
            />
        </form>
    );
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
