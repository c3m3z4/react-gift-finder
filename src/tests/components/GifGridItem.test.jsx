import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />>', () => {
    const title = 'Goku';
    const url = 'https://github.com/goku.jpg';

    test('should match with snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar la  imagen cone l URL y el ALT indicado', () => {
        render(<GifGridItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el titulo del componente', () => {
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
});
