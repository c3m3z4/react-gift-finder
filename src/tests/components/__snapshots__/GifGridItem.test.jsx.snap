// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem />> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://github.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
